File: android/app/src/main/java/vn/team/freechat/socket/SocketRequests.java
package vn.team.freechat.socket;

import com.tvd12.ezyfoxserver.client.EzyClient;
import com.tvd12.ezyfoxserver.client.EzyClients;
import com.tvd12.ezyfoxserver.client.constant.EzyCommand;
import com.tvd12.ezyfoxserver.client.entity.EzyApp;
import com.tvd12.ezyfoxserver.client.entity.EzyZone;
import com.tvd12.ezyfoxserver.client.request.EzyRequest;

import vn.team.freechat.constant.Commands;
import vn.team.freechat.request.GetContactsRequest;
import vn.team.freechat.request.SendSystemMessageRequest;
import vn.team.freechat.request.SendUserMessageRequest;

public final class SocketRequests {

    private SocketRequests() {}

    public static void sendGetContacts() {
        EzyApp app = getApp();
        if(app != null)
            app.send(new GetContactsRequest(0, 50));
    }

    public static void sendGetSuggestContacts() {
        EzyApp app = getApp();
        if(app != null)
            app.send(Commands.SUGGEST_CONTACTS);
    }

    public static void sendSystemMessage(String message) {
        EzyApp app = getApp();
        if(app != null)
            app.send(new SendSystemMessageRequest(message));
    }

    public static void sendUserMessage(long channelId, String message) {
        EzyApp app = getApp();
        if(app != null)
            app.send(new SendUserMessageRequest(channelId, message));
    }

    private static EzyApp getApp() {
        EzyZone zone = getClient().getZone();
        if(zone == null)
            return null;
        return zone.getApp();
    }

    private static EzyClient getClient() {
        EzyClients clients = EzyClients.getInstance();
        return clients.getDefaultClient();
    }
}

File: kotlin/app/src/main/java/vn/team/freechat_kotlin/socket/SocketRequests.kt
package vn.team.freechat_kotlin.socket

import com.tvd12.ezyfoxserver.client.EzyClient
import com.tvd12.ezyfoxserver.client.EzyClients
import com.tvd12.ezyfoxserver.client.entity.EzyApp
import com.tvd12.ezyfoxserver.client.factory.EzyEntityFactory

import java.util.Arrays

import vn.team.freechat_kotlin.constant.Commands
import vn.team.freechat_kotlin.request.GetContactsRequest
import vn.team.freechat_kotlin.request.SendSystemMessageRequest
import vn.team.freechat_kotlin.request.SendUserMessageRequest

class SocketRequests private constructor() {
    companion object {
        fun sendGetContacts() {
            getApp()?.send(GetContactsRequest(0, 50))
        }

        fun sendGetSuggestContacts() {
            getApp()?.send(Commands.SUGGEST_CONTACTS)
        }

        fun sendAddContact(user: String) {
            sendAddContacts(Arrays.asList(user))
        }

        fun sendSearchContacts(keyword: String) {
            val app = getApp()
            if(app != null) {
                val data = EzyEntityFactory.newObjectBuilder()
                    .append("keyword", keyword)
                    .build()
                app.send(Commands.SEARCH_CONTACTS, data)
            }
        }

        fun sendAddContacts(users: Collection<String>) {
            val app = getApp()
            if(app != null) {
                val data = EzyEntityFactory.newObjectBuilder()
                    .append("target", users)
                    .build()
                app.send(Commands.ADD_CONTACTS, data)
            }
        }

        fun sendSystemMessage(message: String) {
            getApp()?.send(SendSystemMessageRequest(message))
        }

        fun sendUserMessage(channelId: Long, message: String) {
            getApp()?.send(SendUserMessageRequest(channelId, message))
        }

        private fun getApp(): EzyApp? {
            val zone = getClient()?.zone ?: return null
            return zone.app
        }

        private fun getClient(): EzyClient? {
            val clients = EzyClients.getInstance()
            return clients.defaultClient
        }
    }
}
